include/openssl/mem.h

```diff
@@ -165,6 +165,24 @@ OPENSSL_EXPORT void CRYPTO_free(void *ptr, const char *file, int line);
 // allocations on free, but we define |OPENSSL_clear_free| for compatibility.
 OPENSSL_EXPORT void OPENSSL_clear_free(void *ptr, size_t len);
 
+// CRYPTO_set_mem_functions is imported to be compatible with OpenSSL 1.1.1.
+// It's used to override the implementation of |OPENSSL_malloc/free/realloc|.
+//
+// Before this import, |OPENSSL_malloc/free/realloc| can be customized by
+// defining |OPENSSL_memory_alloc/free/realloc|.
+// Below warning is also applicable here.
+// -- https://github.com/awslabs/aws-lc/blame/d164f5762b1ad5d4f2d1561fb85daa556fdff5ef/crypto/mem.c#L111-L127
+// This function is only recommended for debug purpose(e.g. track mem usage).
+//
+// One difference is OpenSSL memory functions always receive __FILE__ and __LINE__.
+// https://github.com/openssl/openssl/blame/b7ce611887cfac633aacc052b2e71a7f195418b8/include/openssl/crypto.h#L117-L144
+// But AWS-LC memory functions for now ignore the file and line parameters.
```

Comment:
We should not phrase our documentation about what is imported or changed. "This import" only makes sense in the context of this PR, the documentation should stand on it's own. We can provide the context of this change in this PR.
```suggestion
// CRYPTO_set_mem_functions is used to override the implementation of |OPENSSL_malloc/free/realloc|.
//
// |OPENSSL_malloc/free/realloc| can be customized by implementing |OPENSSL_memory_alloc/free/realloc| or calling
// CRYPTO_set_mem_functions. If  |OPENSSL_memory_alloc/free/realloc| is defined CRYPTO_set_mem_functions will fail.
// All of the warnings for |OPENSSL_malloc/free/realloc| apply to CRYPTO_set_mem_functions:
// -- https://github.com/awslabs/aws-lc/blame/d164f5762b1ad5d4f2d1561fb85daa556fdff5ef/crypto/mem.c#L111-L127
// This function is only recommended for debug purpose(e.g. track mem usage).
// AWS-LC differs from OpenSSL's  CRYPTO_set_mem_functions in that __FILE__ and __LINE__ are not supplied.
```